### PR TITLE
[LLVM] Specify type for the polymorphic derived types

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -9346,8 +9352,11 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
             llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
 
-            llvm::Value* vptr_int_hash = llvm_utils->CreateLoad(llvm_utils->create_gep(llvm_dt, 0));
-            llvm::Value* dt_data = llvm_utils->CreateLoad(llvm_utils->create_gep(llvm_dt, 1));
+            llvm::Type* i64 = llvm::Type::getInt64Ty(context);
```
Comment (gxyd) [inline src/libasr/codegen/asr_to_llvm.cpp:9355]: A nitpicking would be have this line of code outside the loop?

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 2]
```diff
@@ -9433,8 +9442,11 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
```
Comment (gxyd) [inline src/libasr/codegen/asr_to_llvm.cpp:9445]: same for this line to be outside the loop?

**Review by gxyd — APPROVED** [review]
I like the changes overall. Just two nitpick comments, upto you if you want to resolve them @Thirumalai-Shaktivel

**Review by certik — APPROVED** [review]
Thanks. I think @gxyd feedback can be addressed in a subsequent PR.

**Comment by certik** [comment]
Conflicts have to be resolved, then it can be merged.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the reviews!

**Pull request merged**